pkgs/tools/package-management/nix/common.nix

```diff
@@ -216,6 +238,8 @@ self = stdenv.mkDerivation {
   ] ++ lib.optional (stdenv.hostPlatform != stdenv.buildPlatform) "PRECOMPILE_HEADERS=0"
     ++ lib.optional (stdenv.hostPlatform.isDarwin) "PRECOMPILE_HEADERS=1";
 
+  mesonFlags = [ (lib.mesonBool "doc-gen" enableDocumentation) ];
```

Comment:
Rebuilds of all nix versions are probably because of this.

Reply:
We only need to set this if we use meson, no? atLeast226

Reply:
If I set it to null, I get `error: expected a list but found null: null` during evaluation. If I set it to an empty list, it still triggers a rebuild

Reply:
Ah yes, and conditionally merging two attrsets will put it too far away. Probably ok to eat the rebuilds then.